src/containers/edit-profile/password-security-tab/PasswordSecurityTab.tsx

```diff
@@ -1,34 +1,136 @@
-import { confirmPassword, password } from '~/utils/validations/login'
-import { ButtonVariantEnum, InputEnum, SizeEnum, FormValues } from '~/types'
 
-const PasswordSecurityTab = () => {
+import {
+  ButtonVariantEnum,
+  ComponentEnum,
+  ButtonTypeEnum,
+  InputEnum,
+  SizeEnum,
+  FormValues,
+  ErrorResponse,
+  EditProfileTabUserProps
+} from '~/types'
+import {
+  initialValues,
+  validations
+} from '~/containers/edit-profile/password-security-tab/PasswordSecurityTab.constants'
+
+const PasswordSecurityTab: FC<EditProfileTabUserProps> = ({ user }) => {
   const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+
   const [isConfirmOpen, setIsConfirmOpen] = useState(false)
 
-  const { data, handleInputChange, errors, handleBlur, resetData } =
-    useForm<FormValues>({
-      initialValues: {
-        currentPassword: '',
-        password: '',
-        confirmPassword: ''
-      },
-      validations: { password, confirmPassword }
-    })
+  const handleResponse = () => {
+    dispatch(
+      openAlert({
+        severity: snackbarVariants.success,
+        message: 'editProfilePage.profile.successMessage'
+      })
+    )
+  }
+
+  const changePassword = useCallback(
+    (data: { password: string; currentPassword: string }) => {
+      return AuthService.changePassword(user._id, data)
+    },
+    [user._id]
+  )
+
+  const handleResponseError = (error?: ErrorResponse) => {
+    if (error?.code === 'INCORRECT_CREDENTIALS') {
+      handleErrors('password', t('common.errorMessages.samePasswords'))
+    } else {
+      handleErrors('currentPassword', t('common.errorMessages.currentPassword'))
+    }
+  }
+
+  const { loading, fetchData: sendChangedPassword } = useAxios({
+    service: changePassword,
+    onResponse: handleResponse,
+    onResponseError: handleResponseError,
+    fetchOnMount: false
+  })
+
+  const {
+    data,
+    handleSubmit,
+    handleInputChange,
+    errors,
+    handleBlur,
+    resetData,
+    resetErrors,
+    handleErrors
+  } = useForm<FormValues>({
+    onSubmit: async () => {
+      resetErrors()
+      await sendChangedPassword({
+        password: data.password,
+        currentPassword: data.currentPassword
+      })
+    },
+    initialValues: initialValues,
+    validations: validations
+  })
+
+  const {
+    inputVisibility: currentPasswordVisibility,
+    showInputText: showCurrentPassword
+  } = useInputVisibility(errors.currentPassword)
+
+  const { inputVisibility: passwordVisibility, showInputText: showPassword } =
+    useInputVisibility(errors.password)
+
+  const {
+    inputVisibility: newPasswordVisibility,
+    showInputText: showNewPassword
+  } = useInputVisibility(errors.confirmPassword)
+
+  const handlePasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    handleErrors('password', '')
+    handleInputChange('password')(e)
+  }
+
+  const handleCurrentPasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    handleErrors('currentPassword', '')
```

Comment:
merge into one function and pass variables as arguments

Reply:
Done! This functions was unnecessary and duplicated the behavior laid down in useForm (handleInputChange)